### PR TITLE
Query JVM for LIBPATH location

### PR DESCRIPTION
**Pull request by jdekonin**
Signed-off-by: Joe deKoning <joe_dekoning@ca.ibm.com>

### CODE REVIEW

File: test/TestConfig/extraSettings [hunk 1]
```diff
@@ -48,28 +48,11 @@ ifneq (,$(findstring cmprssptrs,$(SPEC)))
...
-else ifneq (,$(findstring 390,$(SPEC)))
-	ARCH_DIR=s390
-else ifneq (,$(findstring linux_x86-64,$(SPEC)))
-	ARCH_DIR=amd64
-else ifneq (,$(findstring linux_x86,$(SPEC)))
-	ARCH_DIR=i386
-endif
-
 ifneq (,$(findstring win,$(SPEC)))
 	JAVA_SHARED_LIBRARIES_DIR:=$(JAVA_BIN)$(D)$(VM_SUBDIR)
```
Comment (DanHeidinga) [inline test/TestConfig/extraSettings]: Is there a reason to keep the windows special case?

Comment (jdekonin) [reply test/TestConfig/extraSettings]: I don't think windows uses LIBPATH at all.  I was planning on doing a search and testing to further cleanup this file before removing the WIP.

**Comment by keithc-ca** [comment]
The tests are still run against Java 8 SDKs where that subdirectory is still present.

**Comment by jdekonin** [comment]
This specifically enables openj9 sanity testing on an AdoptOpenJDK generated AIX Java9 binary.

File: test/TestConfig/extraSettings.mk [hunk 2]
```diff
@@ -71,7 +71,7 @@ ifneq (,$(findstring win,$(SPEC)))
 else
 	JAVA_SHARED_LIBRARIES_DIR:=$(JAVA_LIB_DIR)$(D)$(ARCH_DIR)$(D)$(VM_SUBDIR)
```
Comment (keithc-ca) [inline test/TestConfig/extraSettings.mk]: This definition is not appropriate for Java 9 (the arch directory was removed ~b150).

Comment (keithc-ca) [reply test/TestConfig/extraSettings.mk]: A better solution is to move the test shared libraries out of the VM.

Comment (pshipton) [reply test/TestConfig/extraSettings.mk]: i.e. #805

Comment (pshipton) [reply test/TestConfig/extraSettings.mk]: That said, I don't see why we can't merge this change and leave #805 for another day. This comment doesn't seem to have anything to do with the additional to the LIBPATH below. I also note that while JAVA_SHARED_LIBRARIES_DIR appears wrong for Java 9, the testing is working as expected.

Comment (pshipton) [reply test/TestConfig/extraSettings.mk]: Or to put this another way, I expect this setting for JAVA_SHARED_LIBRARIES_DIR supports the IBM internal b148 build. If we're going to update it, we'll need to do something internal to IBM in order to keep b148 tests working.

**Pull request merged**